### PR TITLE
Update examples to mbed OS 5.3

### PR DESCRIPTION
**Pull request by janjongboom**
@iriark01 

This example broke somewhere between mbed OS releases. I've reworked all examples to use the latest and greatest tech (like mbed Events) and make sure it works again.

### CODE REVIEW

File: docs/3_software.md [hunk 1]
```diff
@@ -10,76 +10,99 @@ Part of the ARM mbed IoT Device Platform is an online compiler, which we will us
...
-1. Click *Import*.
+The program you just imported already contains some boilerplate, including mbed OS and a configuration file. In this configuration file you can configure the pins you're using, after which you can start writing some code.
 
 ## Adding the code
 
-You'll need to create two files; the first one is a configuration file in which you specify which pins you are using.
-Create a new file (right click on the project and select *New File…*), and name it ``config.h``:
+mbed OS comes with a powerful [configuration system](https://docs.mbed.com/docs/mbedmicro-api/en/latest/api/md_docs_config_system.html) that makes it easy to separate configuration and application code. In this application you'll use this to separate the configuration of the LED (cathode, anode or a Grove LED), pins used, and - in the next section - the connectivity method used.
+
+Open ``mbed_app.json`` in the tree and edit the file to reflect your LED chose, and the pins you used to connect the LED and the PIR sensor.
```
Comment (iriark01) [inline docs/3_software.md]: You probably meant "LED choice"

**Comment by iriark01** [comment]
How about we merge it now so that we have the correct code, and then work on prettifying the text?

**Comment by janjongboom** [comment]
@iriark01 Fine by me.

**Comment by janjongboom** [comment]
@iriark01 There's some issues with indentation of lists with inline images. Do you know how to fix those? F.e. https://docs.mbed.com/docs/building-an-internet-connected-lighting-system/en/latest/3_software/

On GH they show nicely.

**Comment by iriark01** [comment]
You need four spaces, not a tab. And probably an extra line break.

**Pull request merged**